tracing.go

```diff
@@ -252,10 +253,26 @@ func ExtractInboundSpan(ctx context.Context, call *InboundCall, headers map[stri
 
 func setPeerHostPort(span opentracing.Span, hostPort string) {
```

Comment:
That's the issue I linked above. I'm getting a bit of push back, but we can always go with a custom tag `peer.hostPort` (won't be portable with other OT tracers if the tag doesn't get standardized).

The good thing about that tag is that when we sample 1 in 1000 spans, setting that tag is no-op, while here we always try to parse the string even though the trace may not be sampled.